### PR TITLE
Envest/remove rows all same

### PR DESCRIPTION
**Pull request by envest**
This PR branches off of https://github.com/greenelab/RNAseq_titration_results/pull/47.

Before z-normalization, it's necessary to check if any rows have all the same values. If they do, NaNs crop up later. The `all_same()` function written in `util/normalization_functions.R` does this check. It returns true if a row has all the same values within a specified numerical tolerance.

This function is used in three files:
- 1-normalize_titrated_data.R
- 1A-detect_differentially_expressed_genes.R
- util/differential_expression_functions.R

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
This is a great example of where using branches in this repo (rather than in your fork) and changing the target branch (https://github.com/greenelab/RNAseq_titration_results/pull/44#issuecomment-889040208) for stacked PRs is important! It is difficult to review this otherwise and I’m going to wait until #47 goes in, as #47 is already touching a lot of files.

**Comment by envest** [comment]
https://github.com/greenelab/RNAseq_titration_results/pull/47 has now been merged 😄

File: util/normalization_functions.R [hunk 1]
```diff
@@ -344,6 +344,24 @@ SinglePlatformNormalizationWrapper <- function(dt, platform = "array",
   return(norm.list)
 }
 
+all_same <- function(x, my_tolerance = 1e-9){
```
Comment (jaclyn-taroni) [inline util/normalization_functions.R:347]: I would consider making this a little bit more descriptive, perhaps something like `check_all_same()`?

**Review by jaclyn-taroni — APPROVED** [review]
LGTM! One comment about the name of the new function, but no need for me to re-review.

**Pull request merged**